### PR TITLE
Fix logic for salvage eligibility

### LINKED ISSUE #1240: [0.45.4 : Bug] Cannot Salvage
### Environment

MekHQ 0.45.4

### Description

Mission Type: Guerrilla Warfare
Salvage Percentage: 100%

During resolution after a battle, all salvageable units were greyed out despite my contract having 100% salvage rights. Lots of error in  error log.

To replicate:
1. Load campaign
2. Select mission #052
3. Select second last scenario, hide and seek
4. Load save game
5. end the game, control battlefield

### Files

[CannotSalvageLog.txt](https://github.com/MegaMek/mekhq/files/3481031/CannotSalvageLog.txt)
[Uploading CannotSalvageCampaign.zip…]()
[Uploading CannotSalvageGame.sav.gz…]()

### PR DESCRIPTION
**Pull request by neoancient**
Currently when determining whether a unit is eligible to be salvaged, the current salvage percentage is calculated with the battle being resolved taken into account, and salvage is disallowed if it meets or exceeds what is allowed by the contract. This means that nothing can be salvaged if the current value (with rounding) is calculated at 100% even if the contract allows 100%. It also allows for the possibility that a unit could take the salvage percentage over the limit if the current value is under the limit before it is chosen.

This PR adjusts the logic to calculate what the salvage percentage would be after selecting the unit and compare that to the limit instead. Salvage is disallowed if it would exceed the limit.

Fixes #1240

### CODE REVIEW

File: MekHQ/src/mekhq/gui/dialog/ResolveScenarioWizardDialog.java [hunk 1]
```diff
@@ -443,6 +443,8 @@ private void initComponents() {
             gridBagConstraints.weightx = 1.0;
             pnlPilotStatus.add(kiaCheck, gridBagConstraints);
             i++;
+
+            kiaCheck.addItemListener(new CheckBoxKIAListener(hitSlider, miaCheck, null));
```
Comment (NickAragua) [inline MekHQ/src/mekhq/gui/dialog/ResolveScenarioWizardDialog.java]: Looks like a bad merge, I don't think we need to be adding two item listeners to that checkbox.

**Comment by NickAragua** [comment]
Also, as a word of warning, this changes the current behavior, which *does* allow you to go over with one unit. 

Which means it'll be hard to salvage an intact grounded dropship now. :(

File: MekHQ/src/mekhq/gui/dialog/ResolveScenarioWizardDialog.java (hunk 1 above)
Comment (neoancient) [reply MekHQ/src/mekhq/gui/dialog/ResolveScenarioWizardDialog.java]: It's in there twice in current master. Which one should it be?

Comment (NickAragua) [reply MekHQ/src/mekhq/gui/dialog/ResolveScenarioWizardDialog.java]: My bad. It should be the top one.

**Comment by neoancient** [comment]
If the ability to go over is expected and intended, then maybe I should close this and just treat 100% as a special case.

**Comment by NickAragua** [comment]
It's definitely expected currently, as it's been that way as long as I can remember. No idea about whether it was intended or not.

**Pull request closed without merging**